### PR TITLE
Adding debug logs

### PR DESCRIPTION
**Pull request by madhawa-gunasekara**
https://wso2.org/jira/browse/DS-1199

### CODE REVIEW

**Comment by madhawa-gunasekara** [comment]
@lafernando Can you please review the logs ?

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java [hunk 1]
```diff
@@ -937,6 +944,9 @@ private Object processPreStoredProcQuery(InternalParamCollection params,
             throw new DataServiceFault(e, FaultCodes.DATABASE_ERROR,
                     "Error in 'SQLQuery.processStoredProcQuery': " + e.getMessage());
         } finally {
+            if (log.isDebugEnabled()){
```
Comment (lafernando) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Fix code formatting.

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/DBInOnlyMessageReceiver.java [hunk 2]
```diff
@@ -45,12 +45,23 @@
...
 		} catch(Exception e) {	
 		    e.printStackTrace();
 			log.error("Error in in-only message receiver", e);
 			msgContext.setProperty(Constants.FAULT_NAME, DBConstants.DS_FAULT_NAME);
 			throw DBUtils.createAxisFault(e);
+		} finally {
+			if (log.isDebugEnabled()) {
+				log.debug("Request processing completed from DSS. Data Service :" +
+				          msgContext.getServiceContext().getName() + ", Operation :" + msgContext.getSoapAction() +
+				          ", ThreadID :" + Thread.currentThread().getId());
```
Comment (lafernando) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/DBInOnlyMessageReceiver.java]: When logging, the format should be "some_message: " + ... notice the colon, and the space after that. This must be consistent, fix in all the places.

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/TLConnectionStore.java [hunk 3]
```diff
@@ -75,6 +78,9 @@ public static void commitNonXAConns() {
             if (!conn.isXA()) {
                 try {
                     conn.commit();
+                    if (log.isDebugEnabled()) {
+                        log.debug("Committing non-XA connection. ThreadID:" + Thread.currentThread().getId());
```
Comment (lafernando) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/TLConnectionStore.java]: Shall we print the "conn" object as well, which will invoke it's "toString", so do please override toString in "DataServiceConnection" classes, to print something suitable, for example, in RDBMS ones, it's Connection objects.

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java [hunk 4]
```diff
@@ -1372,6 +1382,14 @@ private PreparedStatement createProcessedPreparedStatement(int queryType,
                 String dynamicSQL = (String) result[0];
                 currentParamCount = (Integer) result[1];
                 String processedSQL = this.createProcessedQuery(dynamicSQL, params, currentParamCount);
+                if (log.isDebugEnabled()) {
+                    String paramsStr = "";
+                    for (int i = 1; i <= this.getParamCount(); i++) {
+                        paramsStr = paramsStr + params.getParam(i) + ",";
+                    }
+                    log.debug("Starting DB calls for " + processedSQL + " with params: " + paramsStr + ". ThreadID:" +
```
Comment (lafernando) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Put some quotes around the SQL statement in the log.

Comment (madhawa-gunasekara) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: noted, will do the changes.

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java (hunk 1 above)
Comment (madhawa-gunasekara) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: noted, will do the changes.

**Comment by madhawa-gunasekara** [comment]
@lafernando Anjana aiya I have fixed the code.

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/DBInOnlyMessageReceiver.java [hunk 5]
```diff
@@ -45,12 +45,23 @@
...
+			if (log.isDebugEnabled()) {
+				log.debug("Request received to DSS:  Data Service - " + msgContext.getServiceContext().getName() +
+				          ", " + "Operation - " + msgContext.getSoapAction() + ", Request body - " +
+				          msgContext.getEnvelope().toString() + ", ThreadID - " + Thread.currentThread().getId());
+			}
 			DataServiceProcessor.dispatch(msgContext);
-		} catch(Exception e) {	
-		    e.printStackTrace();
+		} catch (Exception e) {
+			e.printStackTrace();
```
Comment (lafernando) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/DBInOnlyMessageReceiver.java]: Please remove the printStackTrace here (I know it was there earlier:)) ..

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/TLConnectionStore.java [hunk 6]
```diff
@@ -63,6 +63,10 @@ public static void commitAll() {
 		for (DataServiceConnection conn : conns.values()) {
 		    try {
 		        conn.commit();
+                if (log.isDebugEnabled()) {
+                    log.debug("Committing connection: " + conn.toString() + ",ThreadID - " +
```
Comment (lafernando) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/TLConnectionStore.java]: After a comma, you've to put a space before the next word, e.g. + ", ThreadId - "... Please fix in all the places.

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/DBInOnlyMessageReceiver.java (hunk 5 above)
Comment (madhawa-gunasekara) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/DBInOnlyMessageReceiver.java]: Noted.

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/TLConnectionStore.java (hunk 6 above)
Comment (madhawa-gunasekara) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/TLConnectionStore.java]: Sure, I will fix those.

**Comment by madhawa-gunasekara** [comment]
Merged.

**Pull request merged**